### DISCUSSION TITLE
Support new React Native new architecture (TurboModules/Fabric)

### DISCUSSION THREAD

**Discussion by BraveEvidence** [Nozbe/WatermelonDB#1348] [Q&A] [answered]
Description

See https://reactnative.dev/docs/new-architecture-library-intro

https://reactnative.dev/blog/2022/06/16/resources-migrating-your-react-native-library-to-the-new-architecture

Use Case

Use new React and React Native features

**Comment by radex** [comment, marked as answer]
WatermelonDB is already new architecture-ready, since its preferred native module is a JSI module, and there's no React UI, so Fabric is irrelevant to WatermelonDB.

There's no advantage of TurboModules over pure JSI modules, so we'll keep the legacy Native Module for now without converting it to a TurboModule (since JSI is preferred, and the old NM is only a fallback)
